File: backend/src/main/java/com/yangshengzhou/lucky_sms/pojo/CourseGrade.java
package com.yangshengzhou.lucky_sms.pojo;

import com.baomidou.mybatisplus.annotation.*;
import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 课程成绩实体类
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("course_grades")
public class CourseGrade extends BaseEntity {
    
    @TableId(type = IdType.AUTO)
    private Integer gradeId;
    
    @TableField("student_id")
    private Integer studentId;
    
    @TableField("assignment_id")
    private Integer assignmentId;
    
    @TableField("usual_score")
    private BigDecimal usualScore;
    
    @TableField("midterm_score")
    private BigDecimal midtermScore;
    
    @TableField("final_score")
    private BigDecimal finalScore;
    
    @TableField("total_score")
    private BigDecimal totalScore;
    
    @TableField("gpa_grade")
    private BigDecimal gpaGrade;
    
    @TableField("grade_level_id")
    private Integer gradeLevelId;
    
    @TableField("make_up_score")
    private BigDecimal makeUpScore;
    
    @TableField("retake_score")
    private BigDecimal retakeScore;
    
    @TableField("review_status_id")
    private Integer reviewStatusId;
    
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    @TableField("review_time")
    private LocalDateTime reviewTime;
    
    @TableField("reviewer_id")
    private Integer reviewerId;
    
    @TableField("review_comment")
    private String reviewComment;
    
    @TableField("status_id")
    private Integer statusId;
    

    @TableField(exist = false)
    private Student student;
    
    @TableField(exist = false)
    private TeachingAssignment assignment;
    
    @TableField(exist = false)
    private GradeLevel gradeLevel;
    
    @TableField(exist = false)
    private ReviewStatus reviewStatus;
    
    @TableField(exist = false)
    private User reviewer;
    
    @TableField(exist = false)
    private StatusType status;
}

File: backend-go/models/course_grade.go
package models

import "time"

type CourseGrade struct {
	BaseEntity
	GradeID        int        `json:"gradeId" gorm:"column:grade_id;primaryKey;autoIncrement"`
	StudentID      int        `json:"studentId" gorm:"column:student_id;index"`
	CourseID       int        `json:"courseId" gorm:"column:course_id;index"`
	SemesterID     int        `json:"semesterId" gorm:"column:semester_id;index"`
	RegularScore   float64    `json:"regularScore" gorm:"column:regular_score;type:decimal(5,2)"`
	FinalScore     float64    `json:"finalScore" gorm:"column:final_score;type:decimal(5,2)"`
	TotalScore     float64    `json:"totalScore" gorm:"column:total_score;type:decimal(5,2)"`
	GradePoint     float64    `json:"gradePoint" gorm:"column:grade_point;type:decimal(3,2)"`
	GradeLevel     string     `json:"gradeLevel" gorm:"column:grade_level;size:10"`
	StatusID       int        `json:"statusId" gorm:"column:status_id"`
	GradedAt       *time.Time `json:"gradedAt" gorm:"column:graded_at"`
	GradedBy       int        `json:"gradedBy" gorm:"column:graded_by"`
	
	Student  *Student  `json:"student,omitempty" gorm:"-"`
	Course   *Course   `json:"course,omitempty" gorm:"-"`
	Semester *Semester `json:"semester,omitempty" gorm:"-"`
}

func (CourseGrade) TableName() string {
	return "course_grades"
}
